flocker/node/_docker.py

```diff
@@ -442,10 +442,18 @@ def _add():
             # stop on this container Docker might well complain it knows
             # not the container of which we speak. To prevent this we poll
             # until it does exist.
-            while not self._blocking_exists(container_name):
-                sleep(0.001)
-                continue
+            self._blocking_wait(5, self._blocking_exists, container_name)
```

Comment:
This and the other new timeout value make me a little uneasy.  Of course, the code being replaced here is "block forever" which could certainly have its own problems.

The downfall of timeouts is that sometimes you hit them.  And the really nasty thing is that you're pretty much guaranteed not to hit them during development (because of course you select a timeout that's long enough).  Then when the code runs in some heavily loaded production environment and things slow down enough... you've got some obscure exception you've never seen before.

I'd really like to see what state the container is in, as reported by docker, when this problem is triggered.  If it reveals that it just takes a little while for docker to get the container into the state we asked for, great - I think we don't need the timeout or we can have a really, really long one.

*Alternatively*, this may be the wrong layer at which to apply a fix.  Perhaps Docker calls simply fail sometimes and there's nothing to be done to make them reliable.  Fortunately this code already executes in the context of a loop.  It's possible the best solution is to let these calls fail sometimes and make sure the way the convergence agent handles those failures is by performing another convergence iteration, in which hopefully the calls will fail.

The main thing to figure out is whether these are really transient failures that do go away after docker catches up or if we're doing something basically wrong that's causing persistent error conditions.